addons/website_mass_mailing/controllers/website_form.py

```diff
@@ -0,0 +1,35 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import json
+
+from odoo.http import request
+from odoo.addons.website.controllers.form import WebsiteForm
+from odoo.exceptions import ValidationError
+from odoo.osv import expression
+
+
+class WebsiteNewsletterForm(WebsiteForm):
+
+    def _handle_website_form(self, model_name, **kwargs):
+        """We override this method to update an existing mailing contact if it exists"""
+        if model_name == 'mailing.contact':
+            model_record = request.env.ref('mass_mailing.model_mailing_contact')
+            searchable_fields = request.env[model_name]._get_searchable_fields()
+
+            domain = expression.OR([
+                [(field, '=', value)]
+                for field, value in kwargs.items()
+                if field in searchable_fields])
+
```

Comment:
just to make me feel better, can you add a check that the domain is not empty ? (if it's empty you can just silently return)